libnmstate/nm/checkpoint.py

```diff
@@ -124,6 +126,21 @@ def create(self):
                 raise NMCheckPointPermissionError(str(e))
             raise NMCheckPointCreationError(str(e))
 
+        GLib.timeout_add(
+            self._timeout * 0.5, self._refresh_checkpoint_timeout, None,
+        )
```

Comment:
This basicly keep refreshing the checkpoint timeout tile exception happens or checkpoint destroyed.

How about we just set the checkpoint timeout as 0 and depend on `CheckPoint.__del__()` to destroy this checkpoint?

Note: `__del__` is different from `__exit__`. The later is for `with` context. The `__dell__` is invoked when class destroyed( including context exit).

Reply:
If we do this and for some unexpected reason nmstate crash then the checkpoint is going to stay there forever. We discussed this and we decided to avoid this situation, right? @tyll @EdDev 

Thanks!

Reply:
Yes, an unlimited checkpoint would be bad.